pkg/controller/worker/firewall_reconcile.go

```diff
@@ -126,6 +146,17 @@ func (a *actuator) ensureFirewallDeployment(ctx context.Context, log logr.Logger
 		deploy.Spec.Template.Spec.LogAcceptedConnections = d.infrastructureConfig.Firewall.LogAcceptedConnections
 		deploy.Spec.Template.Spec.SSHPublicKeys = []string{sshKey}
 
+		if networkAccessType == apismetal.NetworkAccessForbidden {
+			if partition.NetworkIsolation == nil || len(partition.NetworkIsolation.AllowedNetworks.Egress) == 0 {
```

Comment:
Put this into the validation webhook as well to prevent cluster creation altogether